library/Ivoz/Provider/Domain/Service/ResidentialDevice/CheckCompanyType.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Ivoz\Provider\Domain\Service\ResidentialDevice;
+
+use Ivoz\Provider\Domain\Model\Company\CompanyInterface;
+use Ivoz\Provider\Domain\Model\ResidentialDevice\ResidentialDeviceInterface;
+
+class CheckCompanyType implements ResidentialDeviceLifecycleEventHandlerInterface
+{
+    public const PRE_PERSIST_PRIORITY = self::PRIORITY_NORMAL;
+
+    /**
+     * @return array<string,int>
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            self::EVENT_PRE_PERSIST => self::PRE_PERSIST_PRIORITY
+        ];
+    }
+
+    /**
+     * @param ResidentialDeviceInterface $residentialDevice
+     *
+     * @return void
+     */
+    public function execute(ResidentialDeviceInterface $residentialDevice): void
```

Comment:
That info is in the signature already.

```suggestion
    public function execute(ResidentialDeviceInterface $residentialDevice): void
```